components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1719,6 +1772,19 @@ where
 
         let mut meta = self.ctx.store_meta.lock().unwrap();
 
+        if meta.atomic_snap_regions.contains_key(&self.region_id()) {
+            drop(meta);
+            panic!(
+                "{} is applying atomic snapshot during destroying",
+                self.fsm.peer.tag
+            );
+        }
+
+        // It's possible that this region gets a snapshot then gets a stale peer msg.
+        // So the data in `pending_snapshot_regions` should be removed here.
```

Comment:
Is this case also possible for `atomic_snap_regions`?

Reply:
I think it's possible so I add a check in `maybe_destroy` to prevent this case.

Reply:
I think we should move the check here which make the logic closer, and also for cases that calling `destroy_peer` bypass `maybe_destroy`.